src/widgets/direct_connect_dialog.cpp

```diff
@@ -53,10 +55,27 @@ DirectConnectDialog::DirectConnectDialog(NetworkManager *p_net_manager) :
 
 void DirectConnectDialog::onConnectPressed()
 {
+  QString l_hostname = ui_direct_hostname_edit->text();
+  if (!SCHEME_PATTERN.match(l_hostname).hasMatch()) {
+    l_hostname = "tcp://" % l_hostname;
+  }
```

Comment:
what is the purpose of the scheme regex if QUrl will validate the url in the same way?